Extends.md

```diff
@@ -0,0 +1,20 @@
+# Struct Inheritance for WGSL
+
+(TBD)
+
+It’s convenient to be able to express structures as combinations of other structures as in Java/TypeScript.
+
+wgsl-linker currently supports an `extends` syntax for struct inheritance (but `extends` is only lightly
+used upstream).
+
+Combining structures by simple composition is possible in standard wgsl.
+by referencing through a member field. But:
+
+* Using fields of the referenced struct requires an additional field dereference "." in the syntax
+  for each level of indirection.
+* Some abstractions may be hard to express w/o inheritance (since we also don't have interfaces in wgsl).
+
+Issues:
+
+* TBD demonstrate an example or drop the feature.
+* Also, would generics for structs be an alternate way to parameterize composition?
```

Comment:
I like the idea of struct/module "shapes". They are interfaces in a sense, but don't bring in OO ideas and implementation of shapes could be structural rather than explicit. Module shapes in particular would be kind of like rust traits, but instead of operating on a struct, they'd operate on a module, which would become the unit of composition in this extended wgsl. I also like the idea of generic functions and structs though am nervous about what they'd do to compile times based on the inference technique chosen.

Reply:
yah, I'm hoping that we can do a simple form of generics w/o inference, to make things easier.